spec/permutations.cry

```diff
@@ -0,0 +1,49 @@
+// The functions in this module are derived from the consensus-specs:
+// @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#helper-functions
+
+module permutations where
+
+// Check if `value` is a power of two integer
+// NOTE: `n` defines the bit length of `value`
+// @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#is_power_of_two
+is_power_of_two : {n} (fin n, n >= 1) => [n] -> Bit
+is_power_of_two value = (value > 0) && ((value && (value - 1)) == 0)
+
+// Reverse the bits of an integer of lenth `n`
+// NOTE: The Python spec only reverses the low-order bits
+// @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#reverse_bits
+reverse_bits : {n, order} (fin n, n >= 1, fin order, order >= 1, order <= n)
+    => [n] -> [n]
+reverse_bits value = zero # reverse (drop`{n - order} value) // `reverse` is a Cryptol primitive function
+
+// Return the bit-reversed permutation, which is an involution (inverts itself)
+// Standard algorithm: https://en.wikipedia.org/wiki/Bit-reversal_permutation
+// @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#bit_reversal_permutation
+bit_reversal_permutation : {size, len} (fin size, size >= 1, fin len, len >= 2, size >= width (len - 1))
```

Comment:
nit: `fin len` is optional here because we have a finite upper bound on its width. Could remove if you want.

Reply:
Oh, nice point. I'll remove it.